web/html/src/manager/virtualization/nets/network-properties.js

```diff
@@ -0,0 +1,432 @@
+  };
+
+  const onSubmit = () => {
+    // Prepare the model for the submit action
+    let definition = unflattenModel(model, true);
+
+    const toClean = ['interface-selection-type', 'ipv6-enabled', 'virtualport-params-type']
+    toClean.forEach(field => {
+      if (definition[field] != null) {
+        delete definition[field];
+      }
+    });
+
+    // Rename the ipv4def and ipv6def properties
+    ["ipv4", "ipv6"].forEach(prop => {
+      if (definition[`${prop}def`] != null) {
+        definition[prop] = definition[`${prop}def`];
+        delete definition[`${prop}def`];
+      }
+    })
+
+    // Split the DNS hosts names to array
+    if (definition.dns != null && definition.dns.hosts != null) {
+      definition.dns.hosts = definition.dns.hosts.map(host => ({
+        address: host.address,
+        names: host.names.split(','),
+      }))
+    }
+
+    props.submit({
+      actionChain,
+      earliest,
+      definition,
+    });
+  };
+
+  const onDateTimeChanged = (date: Date) => {
+    setEarliest(date);
+  }
+
+  const onActionChainChanged = (newActionChain: ?ActionChain) => {
+    setActionChain(newActionChain);
+  }
+
+  const onModeChanged = () => {
+    const defaultIfaceSelection = FieldsData.getValue(model.type, 'pf', false) ?
+      {'interface-selection-type': 'vf'} : {};
+    setModel(Object.assign({},
+        {
+          name: model.name,
+          type: model.type,
+          autostart: model.autostart,
+        },
+        defaultIfaceSelection,
+      )
+    );
+  }
+
+  const modes = Object.keys(FieldsData.mapping).map(value => (
+    {
+      value,
+      label: FieldsData.getValue(value, 'label', value),
+    }
+  ));
+
+  return (
+    <VirtualizationNetworkDevsApi hostId={props.serverId}>
+    {
+      ({
+        netDevices,
+        messages: netDevicesError,
+      }) => {
+        if (netDevices != null) {
+          return (
+            <Form
+              className="form-horizontal"
+              model={model}
+              onValidate={onValidate}
+              onChange={onChange}
+              onSubmit={onSubmit}
+            >
+              <Messages items={props.messages} />
+              <Panel key="general" title={t('General')} headingLevel="h2">
+                { (props.initialModel || {}).name === undefined
+                  && (
+                  <Text
+                    name="name"
+                    label={t('Name')}
+                    required
+                    invalidHint={t('Can not contain the following characters: /\\')}
+                    labelClass="col-md-3"
+                    divClass="col-md-6"
+                  />)
+                }
+                <Select
+                  labelClass="col-md-3"
+                  divClass="col-md-6"
+                  label={t('Network type')}
+                  name="type"
+                  required
+                  disabled={(props.initialModel || {}).type !== undefined}
```

Comment:
I would recommend either using `typeof foo !== 'undefined'` instead, this configuration is safe now, but if the condition flips in the future, simply changing `!==` to `===` in the current code will create a runtime error.

Reply:
ACK